src/NumberGenerator.hpp

```diff
@@ -8,6 +8,7 @@ enum class GeneratorType
 
 struct NumberGenerator
 {
+    virtual ~NumberGenerator();
```

Comment:
You are missing the implementation of this virtual destructor.
```suggestion
    virtual ~NumberGenerator(){}
```

Reply:
Yes, I was just pushing this change :)